apps/performance-tests/src/unified-selection/HiliteSet.test.ts

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { SnapshotDb } from "@itwin/core-backend";
+import { SchemaContext, SchemaJsonLocater } from "@itwin/ecschema-metadata";
+import { createECSchemaProvider, createECSqlQueryExecutor } from "@itwin/presentation-core-interop";
+import { createHiliteSetProvider, Selectable, Selectables } from "@itwin/unified-selection";
+import { Datasets, IModelName } from "../util/Datasets";
+import { run, RunOptions } from "../util/TestUtilities";
+
+describe("hilite", () => {
+  const additionalElementCount = 5;
+  const additionalFunctionalElement2dCount = 1;
+
+  function createSelection({ fullClassName, start, count }: { fullClassName: string; start: number; count: number }) {
+    const selectables: Selectable[] = [];
+    for (let i = start; i < start + count + additionalElementCount; i++) {
+      selectables.push({ className: fullClassName, id: `0x${i.toString(16)}` });
+    }
+    return Selectables.create(selectables);
+  }
```

Comment:
Not a fan of all those magic numbers... What do they mean?

Reply:
I messed up the starting indices so it was using selecting too many elements. Fixed.

Reply:
The `start` and `step` are still magic and are very fragile. E.g. it's easily possible that an extra element is inserted into an iModel in a future itwinjs-core version and that will result in a hard-to-debug test failure here. Instead, I suggest providing an ECSQL query to select ECInstanceIds that you want to create the `Selectables` from. Of course, that query has to run outside of the test to make sure it's execution is not included in our performance measure.

Reply:
Would we not run into a similar problem with ECSQL queries? For example, if we use a query to select all elements and then try to hilite them in the test. If an additional element is inserted the query would return it too. This is already the case for the functional 2D element iModel that contains an extra element which is not included in the test but would be returned by the query.

Reply:
Nvm, fetched the ids based on `UserLabel`.

Reply:
I actually think that supplying a whole ECSQL query, rather than full class name + user label, would be both - easier to understand and more flexible. So instead of:
```ts
{
  fullClassName: "Generic:PhysicalObject",
  userLabel: "test_element",
}
```

do:
```ts
{
  inputQuery: `SELECT ECInstanceId FROM Generic.PhysicalObject WHERE UserLabel = 'test_element'`,
}
```

I also don't think you need to pick those elements by label and should be able to pick them by class.

Reply:
Supplied the query directly and changed to use user labels only where necessary.